core/drand_beacon_public.go

```diff
@@ -125,8 +124,18 @@ func (bp *BeaconProcess) PublicRandStream(req *drand.PublicRandRequest, stream d
 	proxyReq := &proxyRequest{
 		req,
 	}
-	proxyReq.Metadata = &common.Metadata{
-		BeaconID: bp.group.ID,
+	if req.GetMetadata() != nil {
+		if req.GetMetadata().GetBeaconID() != bp.group.ID {
+			return errors.New("request beaconID does not match group beaconID")
+		}
+		proxyReq.Metadata = req.GetMetadata()
+	} else {
+		bp.state.Lock()
```

Comment:
what is this lock protecting? i don't see why it needs to be in this branch but not the other branch of the if statement?